### PR TITLE
tee output to file, with timestamp on each line

### PR DESCRIPTION
**Pull request by warrenvw**
Resolves https://github.com/StackStorm/st2-packages/issues/499.

Remaining tasks:

 - [ ] ~~Add system info to the log, similar to that provided by `st2-submit-debug-info`.~~
 - [x] Log standard error.

### CODE REVIEW

**Comment by LindsayHill** [comment]
Concept looks good. Presumably need to do something similar for BWC install script?

**Comment by warrenvw** [comment]
Please let me know if I'm headed in the right direction. This seemed the easiest way to add timestamps to each line in the `st2bootstrap-{deb|el6|el7}.sh` scripts. However, we're not logging output of `st2_bootstrap.sh`. As it exists right now, we're also not logging standard error.

**Comment by LindsayHill** [comment]
Seems like stderr would be a kinda important thing to capture here

**Comment by warrenvw** [comment]
Yep, it sure does. It's next on my list. :)

**Comment by warrenvw** [comment]
Instead of writing version info to the install log, I'm beginning to think it makes more sense that we get folks to submit the log using `st2-submit-debug-info` (it ships all logs in `/var/log/st2`)? That seems to include all the required information and more. If the script does too much, maybe we could provide an option that ships just the install log file.

Or, we could do as mentioned before and put the required version info in the log itself. But then we're duplicating efforts.

Which of these two options makes the most sense?

File: scripts/st2_bootstrap.sh [hunk 1]
```diff
@@ -181,5 +187,7 @@ else
 
     echo "Running deployment script for st2 ${VERSION}..."
     echo "OS specific script cmd: bash ${BOOTSTRAP_FILE} ${VERSION} ${RELEASE} ${REPO_TYPE} ${USERNAME} ${PASSWORD}"
-    bash ${BOOTSTRAP_FILE} ${VERSION} ${RELEASE} ${REPO_TYPE} ${DEV_BUILD} ${USERNAME} ${PASSWORD}
+    TS=$(date +%Y%m%dT%H%M%S)
+    sudo mkdir -p /var/log/st2
+    bash ${BOOTSTRAP_FILE} ${VERSION} ${RELEASE} ${REPO_TYPE} ${DEV_BUILD} ${USERNAME} ${PASSWORD} 2>&1 | adddate | sudo tee /var/log/st2/st2-install.${TS}.log
```
Comment (armab) [inline scripts/st2_bootstrap.sh]: Nice trick :+1:

Comment (armab) [reply scripts/st2_bootstrap.sh]: Just please verify if entire script hangs with the correct exit code if something is broken during the main installation.

Comment (warrenvw) [reply scripts/st2_bootstrap.sh]: Good catch. I'll submit a fix for that.

Comment (warrenvw) [reply scripts/st2_bootstrap.sh]: I've verified `exit ${PIPESTATUS[0]}` works as expected.

File: scripts/st2_bootstrap.sh [hunk 2]
```diff
@@ -181,5 +187,8 @@ else
...
+    exit ${PIPESTATUS[0]}
```
Comment (armab) [inline scripts/st2_bootstrap.sh]: Awesome :+1:

**Comment by warrenvw** [comment]
I verified that this works on el6, el7, trusty and xenial.

**Comment by warrenvw** [comment]
I assume that we do not need to add system info to the log, since the log will be part of what's shipped by `st2-submit-debug-info`.

File: scripts/st2_bootstrap.sh (hunk 2 above)
Comment (LindsayHill) [reply scripts/st2_bootstrap.sh]: I can't decide if it's good or bad that I'm learning new Bash tricks here. Probably good.

**Pull request merged**